components/brave_rewards/resources/ui/components/checkout/dialogTitle/style.ts

```diff
@@ -10,5 +10,5 @@ export const StyledHeader = styled.h1`
   font-weight: 500;
   font-family: ${p => p.theme.fontFamily.heading};
   font-size: 22px;
-  color: ${p => p.theme.color.grey800};
+  color: ${p => p.theme.palette.grey800};
```

Comment:
I saw this type of change in `components/brave_new_tab_ui/components/default/settings/index.ts` also; were these things which were broken and were caught here? (if so, awesome!). I see a lot of other references to `theme.color` so I am guessing specific named values like `grey800` were moved under `palette` at some point (versus generic items like `theme.color.text`)

Reply:
`theme.palette` defines the color system that encompasses all the specific colors that will likely be used. It's not technically a theme since they are constant, but it's a convenience property that it's accessible on the theme rather than having to `import palette from brave-ui/palette` everywhere. 

`theme.color` defines functionally-named colors (`contextMenuForeground`, `pageBackground`, etc) whose value changes according to the theme selected (`light`, `dark`, `shieldsLight`, `shieldsDark`, etc). Their values _should_ be from the color system values of the palette.

It's pretty much always been this way, so if anything was the other way round then it was broken and not caught by typescript because of incorrect type declarations or config, though should hopefully be now (not that that's the main purpose of this PR).

Reply:
(If the names are too difficult for devs to remember to differentiate, we could always combine to `theme.color`, but I think it's nice to keep separate as it's nice to remember that `color.x` will change automatically based on theme, and `palette.x` will need to have a dark mode catered to manually in the consumer. However, this architecture is all up for change soon and may just change to css variables anyway.)

Reply:
@bsclifton `theme.palette` is included in the theme here https://github.com/brave/brave-ui/blob/91bf96d8063affc153e085cb7b5ed659e1eba95c/src/theme/brave-default.ts#L10